Gems/Atom/RPI/Code/Source/RPI.Public/Pass/CopyPass.cpp

```diff
@@ -684,5 +702,138 @@ namespace AZ
             m_copyItemSameDevice = copyDesc;
         }
 
+        RHI::Ptr<Query> CopyPass::GetQuery(ScopeQueryType queryType, int scopeIndex)
+        {
+            uint32_t typeIndex = static_cast<uint32_t>(queryType);
```

Comment:
May want to add an assert if scopeIndex is greater than the size of m_perScopeQueries

Reply:
I would argue we can do without, as this is a `private` method and only called locally with hard-coded values `0` and `1`.